Sming/Core/Network/TcpConnection.cpp

```diff
@@ -42,16 +42,17 @@ bool TcpConnection::connect(const String& server, int port, bool useSsl, uint32_
 	ip_addr_t addr;
 
 	this->useSsl = useSsl;
-#ifdef ENABLE_SSL
+
```

Comment:
I do wonder about having two variables (`useSsl` and `sslOptions`). We could simplify the entire interface for this by passing an `Ssl::Options*` structure instead of these two parameters; if it's null, we're not using SSL. (See above notes regarding `Ssl::Options`.)

Reply:
> useSsl

Probably that should stay in TcpConnection. The rest should be moved to a struct as you have suggested.